src/Microsoft.VisualBasic/tests/OperatorsTests.cs

```diff
@@ -5456,11 +5456,25 @@ private static void AssertEqual(object expected, object actual)
         {
             if (expected is double expectedDouble && actual is double actualDouble)
             {
-                Assert.Equal(expected.ToString(), actual.ToString());
```

Comment:
Why are you comparing equality using strings? That will not do what you expect for inputs that only differ in the 16th or 17th digit?

This should probably be doing:
```csharp
if (double.IsNaN(expectedDouble))
{
    Assert.True(double.IsNaN(actualDouble));   
}
else
{
    Assert.Equal(BitConverter.DoubleToInt64Bits(expectedDouble), BitConverter.DoubleToInt64Bits(actualDouble));
}
```